src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -332,6 +338,9 @@ namespace ts.server.typingsInstaller {
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
+
+                    this.writePackageVersion(cachePath);

```

Comment:
Each typing's version is stored in its package.json so that's where I'm looking. I think it's already pretty granular. I'm not sure if typesRegistry could add more value here -- I need the version installed on the machine, not what's available on npm.